catalog/video/catalog-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/Mpeg2PictureType.java

```diff
@@ -13,8 +13,38 @@
  */
 package org.codice.alliance.video.stream.mpegts.netty;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
 public enum Mpeg2PictureType {
 
-    INTRA_CODED, PREDICTIVE_CODED, BIDIRECTIONALLY_PREDICTIVE_CODED
+    INTRA_CODED(1), PREDICTIVE_CODED(2), BIDIRECTIONALLY_PREDICTIVE_CODED(3);
+
+    private static final Map<Long, Mpeg2PictureType> LOOKUP = new HashMap<>();
+
+    static {
+        for(Mpeg2PictureType mpeg2PictureType : values()) {
+            LOOKUP.put(mpeg2PictureType.h262HeaderValue, mpeg2PictureType);
+        }
+    }
+
+    /**
+     * This is the value that appears in an H262 encoded stream.
+     */
+    private final long h262HeaderValue;
+
+    Mpeg2PictureType(long h262HeaderValue) {
+        this.h262HeaderValue = h262HeaderValue;
+    }
+
+    /**
+     * Find the Mpeg2PictureType that corresponds to the value encoded in an H262 header.
+     * @param h262HeaderValue header value
+     * @return Mpeg2PictureType or empty
+     */
+    public static Optional<Mpeg2PictureType> fromH262HeaderValue(long h262HeaderValue) {
+        return Optional.ofNullable(LOOKUP.getOrDefault(h262HeaderValue, null));
```

Comment:
That was what I did at first, but I thought getOrDefault better communicated my intentions.